docs/source/upgrade.rst

```diff
@@ -0,0 +1,747 @@
+
+
+Global configurations
+---------------------
+
+.. _upgrade-train-mode:
+
+Training mode is configured by a thread-local flag
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, the concept of *training mode* is added.
+It is represented by a thread-local flag ``chainer.config.train``, which is a part of :ref:`the unified configuration <configuration>`.
+When ``chainer.config.train`` is true, functions of Chainer run in the training mode, and otherwise they run in the test mode.
+For example, :class:`~links.BatchNormalization` and :func:`~functions.dropout` behave differently in each mode.
+
+In Chainer v1, such a behavior was configured by the ``train`` argument of each function.
+**This train argument has been removed in Chainer v2.**
+If your code is using the ``train`` argument, you have to update it.
+In most cases, what you have to do is just removing the ``train`` argument from any function calls.
+
+.. admonition:: Example
+
+   Consider the following model definition and the code to call it in test mode written for Chainer v1.
+
+   .. code-block:: py
+
+      # Chainer v1
+      import chainer.functions as F
+
+      class MyModel(chainer.Link):
+          ...
+
+          def __call__(self, x, train=True):
+              return f(F.dropout(x, train=train))
+
+      m = MyModel(...)
+      y = m(x, train=False)
+
+   In Chainer v2, it should be updated into the following code:
+
+   .. code-block:: py
+
+      # Chainer v2
+      import chainer.functions as F
+
+      class MyModel(chainer.Link):
+          ...
+
+          def __call__(self, x):
+              return f(chainer.functions.dropout(x))
+
+      m = MyModel(...)
+      with chainer.using_config('train', False):
+          y = m(x)
+
+.. _upgrade-configurations:
+
+Configurations are added and replace some of existing global flags
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are many global settings moved to :ref:`the unified configuration <configuration>` other than the training mode.
+Following is the complete list of the configuration entries that have corresponding features in Chainer v1.
+
+``chainer.config.debug``
+    It is corresponding to the debug mode in Chainer v1, which was configured by :func:`set_debug` and extracted by :func:`is_debug`.
+    These functions are also available in Chainer v2, so you basically do not need to update the code related to the debug mode.
+``chainer.config.deterministic``
+    It is corresponding to the ``deterministic`` argument of some convolution functions in Chainer v1.
+    **This argument has been removed since Chainer v2.**
+    If you are using this argumetn, you have to use the ``chainer.config.deterministic`` flag instead to change the behavior of the convolution functions.
+``chainer.config.enable_backprop``
+    It is corresponding to *the backprop mode* in Chainer v1, which was configured by :func:`no_backprop_mode` and :func:`force_backprop_mode`.
```

Comment:
Should we emphasize that `force_backprop_mode` and `no_backprop_mode` remains in v2?